.circleci/config.yml

```
@@ -109,15 +109,18 @@ jobs:
           command: |
             pids=()
 
-            pnpm nx-cloud record -- nx format:check --base=$NX_BASE --head=$NX_HEAD &
-            pids+=($!)
+            function run_command() {
+              local command=$1
+              $command &
+              pids+=($!) 
+            }
 
-            pnpm nx run-many -t check-imports check-commit check-lock-files check-codeowners documentation --parallel=1 --no-dte &
-            pids+=($!)
+            run_command "pnpm nx-cloud record -- nx format:check --base=$NX_BASE --head=$NX_HEAD"
 
-            (pnpm nx affected --targets=lint,test,build --base=$NX_BASE --head=$NX_HEAD --parallel=3 &&
-            pnpm nx affected --target=e2e --base=$NX_BASE --head=$NX_HEAD --parallel=1) &
-            pids+=($!)
+            run_command "pnpm nx run-many -t check-imports check-commit check-lock-files check-codeowners documentation --parallel=1 --no-dte"
+
+            run_command "pnpm nx affected --targets=lint,test,build --base=$NX_BASE --head=$NX_HEAD --parallel=3"
+            run_command "pnpm nx affected --target=e2e --base=$NX_BASE --head=$NX_HEAD --parallel=1"
```

The E2E commands should not run in parallel to the others. Since they are very expensive, we only run them if lint, test and build passed.

```suggestion
            run_command "(pnpm nx affected --targets=lint,test,build --base=$NX_BASE --head=$NX_HEAD --parallel=3 && pnpm nx affected --target=e2e --base=$NX_BASE --head=$NX_HEAD --parallel=1)"
```

---

I took the liberty to update the necessary change so that we can merge it. Hope you don't mind.